### DISCUSSION TITLE
Force bluetooth disconnect?

### DISCUSSION THREAD

**Discussion by clarkbremer** [pschatzmann/ESP32-A2DP#191] [Q&A] [answered]
Is there a function to force the bluetooth to disconnect?  When switching from one device to another, it would be helpful if I didn't have to initiate the disconnect from one device, then initiate the connect from the other.

**Comment by pschatzmann** [comment, marked as answer]
I am sorry, I am confused. It is unclear to me what you are trying to do ? 
Is this for a Bluetooth Receiver or Bluetooth Sender ?
If it is for a Bluetooth Receiver should anybody be able to hijack your bluetooth connection ?

**Comment by clarkbremer** [comment]
Yes, I was unclear.  This would be for an A2DP Sink in an automotive setting.  

Here's a scenario:  Driver and passenger are in the car, and the bluetooth is paired to the driver's phone, since she was last to use it.  Now the passenger wants to take control of the music.  He can't connect until the driver first digs out her phone, and disconnects.  If there were a "disconnect bluetooth" button on the front panel of my device, then the passenger could press that button, and then connect his phone.  Driver does not need to do anything.

**Comment by pschatzmann** [comment]
I think the most natural thing would be to disconnect from the phone.
If you want to do this from the esp32 you can call the disconnect method. I suggest that you have a look the the [class documentation](https://pschatzmann.github.io/ESP32-A2DP/html/annotated.html)

**Comment by clarkbremer** [comment]
Not sure how I missed that.  😵‍💫 Was looking in `BluetoothA2DPSink.h`.  Works exactly how I want.

**Comment by pschatzmann** [comment]
Cool

**Comment by Alistair1231** [comment]
Would it be possible to allow the hijacking of the connection?
So just always disconnect the current device, once another device tries to establish a connection?

**Comment by pschatzmann** [comment]
I guess this might work. Just try to make it discoverable and connectable

**Reply by Alistair1231** [reply]
mmh. Interesting. I tried that, but it doesn't seem to work. vscode shows issues, however that is because I didn't integrate ESP-IDF into it and run the command manually. The code builds and runs without issues.
At first I just set the connectable from `false` to `true` in your code:


<img src="https://github.com/pschatzmann/ESP32-A2DP/assets/26165288/61c6f6ee-6166-44eb-9fcb-d65fa3a1b8af" width=400px/>

After that didn't work I kept it and simply added the method which I found in the [docs](https://docs.espressif.com/projects/esp-idf/en/v4.3/esp32/api-reference/bluetooth/esp_gap_bt.html?highlight=esp_bt_gap_set_scan_mode#_CPPv424esp_bt_gap_set_scan_mode24esp_bt_connection_mode_t23esp_bt_discovery_mode_t)
<img src="https://github.com/pschatzmann/ESP32-A2DP/assets/26165288/4c43602a-0993-4d7c-9ab3-de8181e0736e" width=600px/>
![image](https://github.com/pschatzmann/ESP32-A2DP/assets/26165288/18f32b0f-11e0-4dd7-aac9-824b528daec3)

However when I try to connect it seems to just time out. Do you know what I am doing wrong?
Thanks :)

**Comment by pschatzmann** [comment]
No, but that's what the logging is for!
